secretsauce/apps/portal/models.py

```diff
@@ -113,3 +120,37 @@ class ConstraintParameterRelationship(models.Model):
         on_delete=models.CASCADE,
     )
     coefficient = models.FloatField()
+
+class Schema(models.Model):
+    """List of headers found in a DataBlock"""
+    data_block = models.OneToOneField(
+        DataBlock,
+        on_delete=models.CASCADE,
+        related_name='schema',
+    )
+
+class Header(models.Model):
+    schema = models.ForeignKey(
+        Schema,
+        on_delete=models.CASCADE,
+        related_name='headers',
+    )
+    item_id = models.IntegerField()
+
+class ItemList(models.Model):
+    """List of items obtained from cost sheet."""
+    project = models.OneToOneField(
+        Project,
+        on_delete=models.CASCADE,
+    )
+
+class Item(models.Model):
+    """Item obtained from Cost Sheet"""
+    item_list = models.ForeignKey(
+        ItemList,
+        on_delete=models.CASCADE,
+        related_name='items',
+    )
+    name = models.CharField(max_length=200)
+    item_id = models.IntegerField()
+    cost = models.FloatField(blank=True)
```

Comment:
I find this whole idea of storing an array via two models kinda odd and redundant but I guess that is the way to go for what django fields provide. I know specifically for postgres, there is an Arrayfield for django but considering the extra work to set up locally, I think it is fine to not do it. Let rms settle it themselves afterwards

Reply:
Also, so when creating constraints, we will use these items now? What if there is not cost sheet provided? Where would the items come from?

Reply:
For constraint creation, decision variables should come from HeaderList, from the DataBlock. The item IDs from HeaderList can be cross-referenced with ItemList for their names and/or costs (items which appear on the ItemList and not HeaderList will also be available for cost data). If ItemList is not available, we will only present them with the item IDs from HeaderList.